### PR TITLE
[WIP] Add integration tests; update dockerfile

### PR DESCRIPTION
**Pull request by sengineer0**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -49,11 +49,13 @@ RUN apt-get -qq -y update && \
...
     gnupg1 \
     curl \
+    wget \
     ca-certificates \
     lsb-release && \
     release=`lsb_release -sc` && \
     # Add repo for MongoDB
-    curl "https://www.mongodb.org/static/pgp/server-${MONGODB_VERSION_REPO}.asc" | apt-key add - && \
+    wget "https://www.mongodb.org/static/pgp/server-${MONGODB_VERSION_REPO}.asc" -O mongo-server-${MONGODB_VERSION_REPO}.asc && \
+    apt-key add mongo-server-${MONGODB_VERSION_REPO}.asc && \
```
Comment (newgene) [inline Dockerfile]: @sengineer0 @remoteeng00 why the original line does not work? Seems like we should use either curl or wget in this Dockerfile, but no need to use both?

**Comment by newgene** [comment]
@remoteeng00 @sengineer0 we should not change the existing `files` folder, because it's for setting up a fresh new hub without any data source.

Instead, I think you can put your "demohub" files under `/demohub` subfolder, then modify `Dockerfile` and `Makefile` to use `make demohub` to setup the demohub container, just like other `make *` commands.

Then the corresponding tests files can be put under `/tests/hubapi/demohub` subfolder.

File: Dockerfile (hunk 1 above)
Comment (remoteeng00) [reply Dockerfile]: Oh, It's return `HTTP/1.1 301 Moved Permanently` so we get empty response, I've changed back to `curl` command.

File: Dockerfile [hunk 2]
```diff
@@ -49,11 +51,12 @@ RUN apt-get -qq -y update && \
     apt-get install -y --no-install-recommends \
     gnupg1 \
     curl \
+    wget \
```
Comment (newgene) [inline Dockerfile]: wget can be removed now

Comment (newgene) [inline Dockerfile]: wget can be removed now

File: Dockerfile [hunk 3]
```diff
@@ -147,6 +152,10 @@ RUN curl -LO \
     && ln -s /usr/local/cerebro-${CEREBRO_VERSION} /usr/local/cerebro \
     && rm -rf /tmp/cerebro*
 
+# ES plugins
+RUN /usr/share/elasticsearch/bin/elasticsearch-plugin install --batch repository-s3
+RUN echo $AWS_ACCESS_KEY | /usr/share/elasticsearch/bin/elasticsearch-keystore add --stdin s3.client.default.access_key
+RUN echo $AWS_SECRET_KEY | /usr/share/elasticsearch/bin/elasticsearch-keystore add --stdin s3.client.default.secret_key
```
Comment (newgene) [inline Dockerfile]: @remoteeng00 because we use this `Dockerfile` to build multiple varieties of biothings hub/studio image, not just for demohub, see `Makefile`. So let's keep the default behavior unchanged, you can either:

* Use an ARG variable to controle the demohub specific installs
* Create a separate Dockerfile under demohub folder, leave this one unchanged

This applies for all your changes below to setup a FTP server.

Comment (newgene) [inline Dockerfile]: @remoteeng00 because we use this `Dockerfile` to build multiple varieties of biothings hub/studio image, not just for demohub, see `Makefile`. So let's keep the default behavior unchanged, you can either:

* Use an ARG variable to controle the demohub specific installs
* Create a separate Dockerfile under demohub folder, leave this one unchanged

This applies for all your changes below to setup a FTP server.

File: ansible_playbook/configure.yml [hunk 4]
```diff
@@ -163,3 +163,12 @@
...
 
+################################################
+# Update FTP server config
+################################################
+- name: es backups data directory (repo type fs)
+  file: path=/var/ftp mode=0777 state=directory owner=nobody group=nogroup
+
+- name: update FPT server config
+  template: mode=0644 src=templates/vsftpd.conf dest=/etc/vsftpd.conf
+
```
Comment (newgene) [inline ansible_playbook/configure.yml]: as mentioned above, remove this as the default install, only useful for demohub

Comment (newgene) [inline ansible_playbook/configure.yml]: as mentioned above, remove this as the default install, only useful for demohub

File: files/biothings_studio/bin/hub.py [hunk 5]
```diff
@@ -83,4 +83,4 @@
     logging.info("Hub DB backend: %s" % biothings.config.HUB_DB_BACKEND)
     logging.info("Hub database: %s" % biothings.config.DATA_HUB_DB_DATABASE)
 
-    server.start()
```
Comment (newgene) [inline files/biothings_studio/bin/hub.py]: @remoteeng00 don't know why this change (also a few files below). My linter `flake8` will report an error for not having an empty line at the end.

Comment (newgene) [inline files/biothings_studio/bin/hub.py]: @remoteeng00 don't know why this change (also a few files below). My linter `flake8` will report an error for not having an empty line at the end.

File: tests/hubapi/demohub/bin/ssh_host_key.pub [hunk 6]
```diff
@@ -0,0 +1 @@
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCz5DGnPPVBST8h8+j7nfJ2kGEy0H/rJu1s5iORjWUHMyySZ8c6yQds4eiharqEVemNT/uO8ae06GWD3MP41am8NdScC5Xs97bLbADNn3rw+NHPLD31VmAkWRnivJLDX3tGX1bcygTDs15qilIv+umZRHxihcCBfLiwri7+abdFtmdnJWP8A6Hh2zRHsVCr7QT2IsHMPy8n1bAP8Bk+5NRIHU1eqprLIURnm1cKIIQMXrCTYtZfkmuM9mB639F2/v407G2ZJNyGeqBxgdJK3uZxwFEPWQtfK0urOvzCk5SW8XvgU7rG2nwO0z9dcpV8o1QLxcpSeX4rt/R8BnmAHOe3
```
Comment (newgene) [inline tests/hubapi/demohub/bin/ssh_host_key.pub]: @remoteeng00 should above two files be committed?

Comment (newgene) [inline tests/hubapi/demohub/bin/ssh_host_key.pub]: @remoteeng00 should above two files be committed?

File: tests/hubapi/mvcgi/README [hunk 7]
```diff
@@ -9,3 +9,4 @@ HUB_API_HOST=http://localhost:7080 py.test
 A vanilla Biothings Studio is expected with these tests:
 $ docker pull biothings/biothings-studio
 $ docker run --rm --name test_studio -p 8080:8080 -p 7022:7022 -p 7080:7080 -p 9001:9000 -p 8000:8000 -d biothings/biothings-studio
+
```
Comment (newgene) [inline tests/hubapi/mvcgi/README]: @remoteeng00 should all your changes to this `mvcgi` folder moved to `demohub` folder? and leaves mvcgi folder unchanged?

Comment (newgene) [inline tests/hubapi/mvcgi/README]: @remoteeng00 should all your changes to this `mvcgi` folder moved to `demohub` folder? and leaves mvcgi folder unchanged?

File: tests/hubapi/demohub/bin/ssh_host_key.pub (hunk 6 above)
Comment (sengineer0) [reply tests/hubapi/demohub/bin/ssh_host_key.pub]: sorry, this file is added by me, I forgot to add it to .gitignore
It should be ignored

File: README.md [hunk 8]
```diff
@@ -52,3 +52,24 @@ the new version.
...
+
+# update README: How to run integration tests
+
+Step 1: Install Docker, tavern-ci, pytest
+
+Step 2: Run `copy tests/hubapi/demohub/biothing_studio/config.py.example tests/hubapi/demohub/biothing_studio/config.py`, edit file `config.py` (fill: AWS_ACCESS_KEY, AWS_SECRET_KEY)
+
+Step 3: Run `copy tests/hubapi/demohub/testcases/config.yaml.example tests/hubapi/demohub/testcases/config.yaml`, edit file `config.yaml`
+
+Step 4: Run `copy .env.example .env.local`, edit file `.env.local`  (fill: AWS_ACCESS_KEY, AWS_SECRET_KEY)
```
Comment (newgene) [inline README.md:84]: @remoteeng00 please also include the details like what s3 buckets need to be created and what permission is needed for the given AWS_ACCESS_KEY and AWS_SECRET_KEY.

Comment (remoteeng00) [reply README.md:84]: It's udpated.
